classic-code-snippets.php

```diff
@@ -0,0 +1,243 @@
+		'singular_name'         => __( 'Code Snippet', 'ccs-code-snippet' ),
+		'menu_name'             => __( 'Code Snippets', 'ccs-code-snippet' ),
+		'name_admin_bar'        => __( 'Code Snippet', 'ccs-code-snippet' ),
+		'add_new'               => __( 'Add New', 'ccs-code-snippet' ),
+		'add_new_item'          => __( 'Add New Code Snippet', 'ccs-code-snippet' ),
+		'new_item'              => __( 'New Code Snippet', 'ccs-code-snippet' ),
+		'edit_item'             => __( 'Edit Code Snippet', 'ccs-code-snippet' ),
+		'view_item'             => __( 'View Code Snippet', 'ccs-code-snippet' ),
+		'all_items'             => __( 'All Code Snippets', 'ccs-code-snippet' ),
+		'search_items'          => __( 'Search Code Snippets', 'ccs-code-snippet' ),
+		'parent_item_colon'     => __( 'Parent Code Snippets:', 'ccs-code-snippet' ),
+		'not_found'             => __( 'No Code Snippets found.', 'ccs-code-snippet' ),
+		'not_found_in_trash'    => __( 'No Code Snippets found in Trash.', 'ccs-code-snippet' ),
+		'featured_image'        => __( 'Code Snippet Cover Image', 'ccs-code-snippet' ),
+		'set_featured_image'    => __( 'Set cover image', 'ccs-code-snippet' ),
+		'archives'              => __( 'Code Snippet archives', 'ccs-code-snippet' ),
+		'filter_items_list'     => __( 'Filter Code Snippets list', 'ccs-code-snippet' ),
+		'items_list_navigation' => __( 'Code Snippets list navigation', 'ccs-code-snippet' ),
+		'items_list'            => __( 'Code Snippets list', 'ccs-code-snippet' ),
+	);
+
+	register_post_type(
+		'ccs_code_snippet',
+		array(
+			'labels'       => $labels,
+			'public'       => true,
+			'menu_icon'    => 'dashicons-editor-code',
+			'show_in_menu' => true,
+			'supports'     => array( 'title' ),
+		)
+	);
+}
+add_action( 'init', 'ccs_create_custom_post_type' );
+
+/**
+ * Register Code Snippet metabox.
+ *
+ * @return void
+ */
+function ccs_register_meta_boxes() {
+	add_meta_box(
+		'ccs-code-snippets-metabox',
+		__( 'Code Snippet Details', 'ccs-code-snippet' ),
+		'ccs_display_callback',
+		'ccs_code_snippet'
+	);
+}
+add_action( 'add_meta_boxes', 'ccs_register_meta_boxes' );
+
+/**
+ * Metabox callback with HTML.
+ *
+ * @param mixed $post WP_Post Post object.
+ * @return void
+ */
+function ccs_display_callback( $post ) {
+	$ccs_code_snippet = get_post_meta( get_the_ID(), 'ccs_code_snippet', true );
+	?>
+	<p class="ccs-meta-options">
+		<label for="ccs_code_snippet"><?php esc_attr_e( 'Insert snippet in post/page using', 'ccs-code-snippet'); ?></label>
+		<input id="ccs_code_snippet" type="text" name="ccs_code_snippet" value="<?php echo '[css_snippets id=' . get_the_ID() . ']'; ?>" />
+	</p>
+	<p class="ccs-meta-options">
+		<label for="ccs_code_snippet screen-reader-text"><?php echo esc_attr( 'Code Snippet', 'ccs-code-snippet'); ?></label><br>
+		<textarea id="ccs_code_snippet" rows="20" class="widefat" name="ccs_code_snippet"><?php echo esc_attr( $ccs_code_snippet ); ?></textarea>
```

Comment:
My bad. I see that I left both input fields with the same name. That conflicts with the metadata as it is in a singular form. I will make the html ammends.

Reply:
Fixed